CMakeLists.txt

```diff
@@ -77,7 +56,7 @@ ENDIF (UNIX)
 SET (CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules/")
 
 # hdf5
-# FIND_PACKAGE (HDF5) Do no more try to find HDF5 because it looks to be too much different across different platforms/versions.
+# FIND_PACKAGE (HDF5) is not used because we need debug and release versions of hdf5 for windows build environments. The cmake module does not provide both debug and release library cmake variables.
```

Comment:
The windows user could provide manually the variable with the debug version  of the library. I suppose it is only in a development mode (not for production). In the present situation it is impossible to automatically let cmake detect the hdf5 libraries to automatically build a software on different systems.

Reply:
I agree : this is only in a development mode.
And consequently what your propose should work.
However, I cannot find a general approach which solve all cases:
- only debug mode on windows : findHDF5 for include and lib. Nothing (or same) in debug lib cmake variable?
- release and debug on windows : findHDF5 for include and release lib. debug lib cmake variable for debug.
- release or debug on linux : findHDF5 for include and lib (debug or release).

Anyway, your proposal looks better then the current approach.
I will probably implement something like that. I still have your PR opened to copy this kind of stuff.

Reply:
Another hint. According to the [find module](https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/FindHDF5.cmake) file line 846 it will search for "THIS_LIBRARY_SEARCH_DEBUG" files.. You should use the target hdf5::hdf5 instead of the variables and check the linked libraries.